app/views/admin/users/me.html.erb

```diff
@@ -15,7 +15,11 @@
   <% end %>
   <%= form.actions %>
   <p>
-    <%= t('pageflow.admin.users.me.delete_account_hint') %>
-    <%= link_to(t('pageflow.admin.users.me.delete_account'), delete_me_admin_users_path, :class => 'delete_account') %>
+    <% if Pageflow.config.authorize_user_deletion.call == true %>
+      <%= t('pageflow.admin.users.me.delete_account_hint') %>
+        <%= link_to(t('pageflow.admin.users.me.delete_account'), delete_me_admin_users_path, :class => 'delete_account') %>
+    <% else %>
+        <%= Pageflow.config.authorize_user_deletion.call %>
+    <% end %>
```

Comment:
Let's move this to a helper rendering a partial. That way we can also avoid calling the lambda multiple times as it can potentially perform expensive queries. Also let's apply different class names to the surrounding paragraph tag in each case. That way we can style the message to include a warning icon etc when deleting is forbidden.